Plugins/ExaTrkX/src/TorchEdgeClassifier.cpp

```diff
@@ -19,9 +20,23 @@ namespace Acts {
 
 TorchEdgeClassifier::TorchEdgeClassifier(const Config& cfg,
                                          std::unique_ptr<const Logger> _logger)
-    : m_logger(std::move(_logger)), m_cfg(cfg) {
+    : m_logger(std::move(_logger)),
+      m_cfg(cfg),
+      m_device(torch::Device(torch::kCPU)) {
   c10::InferenceMode guard(true);
   m_deviceType = torch::cuda::is_available() ? torch::kCUDA : torch::kCPU;
+  if (m_deviceType == torch::kCPU) {
+    ACTS_INFO("Running on CPU...");
+  } else {
+    if (cfg.deviceID >= 0 &&
+        static_cast<std::size_t>(cfg.deviceID) < torch::cuda::device_count()) {
+      ACTS_DEBUG("GPU device " << cfg.deviceID << " is being used.");
+      m_device = torch::Device(torch::kCUDA, cfg.deviceID);
+    } else {
+      ACTS_FATAL("GPU device " << cfg.deviceID << " not available. ");
```

Comment:
I think we should not use `ACTS_FATAL` and then continue exectuion. We could
* throw an exception
* Make it a `ACTS_WARNING` and fall back to CPU (I think I would prefer this slightly)

Reply:
done